Find a functionally equivalent code:android.app.Notification.Builder.buildUnstyled:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * Apply the unstyled operations and return a new {@link Notification} object.
 * @hide
 */

@@ -4,10 +4,11 @@
     n.icon = mSmallIcon;
     n.iconLevel = mSmallIconLevel;
     n.number = mNumber;
-    n.contentView = makeContentView();
+    n.color = sanitizeColor();
+    setBuilderContentView(n, makeContentView());
     n.contentIntent = mContentIntent;
     n.deleteIntent = mDeleteIntent;
     n.fullScreenIntent = mFullScreenIntent;
@@ -13,41 +14,47 @@
     n.largeIcon = mLargeIcon;
     n.sound = mSound;
     n.audioStreamType = mAudioStreamType;
+    n.audioAttributes = mAudioAttributes;
     n.vibrate = mVibrate;
     n.ledARGB = mLedArgb;
     n.ledOnMS = mLedOnMs;
     n.ledOffMS = mLedOffMs;
     n.defaults = mDefaults;
     n.flags = mFlags;
-    n.bigContentView = makeBigContentView();
+    setBuilderBigContentView(n, makeBigContentView());
+    setBuilderHeadsUpContentView(n, makeHeadsUpContentView());
     if (mLedOnMs != 0 || mLedOffMs != 0) {
         n.flags |= FLAG_SHOW_LIGHTS;
     }
     if ((mDefaults & DEFAULT_LIGHTS) != 0) {
         n.flags |= FLAG_SHOW_LIGHTS;
     }
-    if (mKindList.size() > 0) {
-        n.kind = new String[mKindList.size()];
-        mKindList.toArray(n.kind);
-    } else {
-        n.kind = null;
-    }
+    n.category = mCategory;
+    n.mGroupKey = mGroupKey;
+    n.mSortKey = mSortKey;
     n.priority = mPriority;
     if (mActions.size() > 0) {
         n.actions = new Action[mActions.size()];
         mActions.toArray(n.actions);
     }
+    n.visibility = mVisibility;
+    if (mPublicVersion != null) {
+        n.publicVersion = new Notification();
+        mPublicVersion.cloneInto(n.publicVersion, true);
+    }
+    // Note: If you're adding new fields, also update restoreFromNotitification().
     return n;
 }

